src/test/java/uk/gov/pay/ledger/util/pagination/PaginationBuilderTest.java

```diff
@@ -138,4 +141,27 @@ public void shouldNotGenerateNextLink_whenLimitTotalIsSetAndCountIsLessThanPageS
         builder = builder.buildResponse();
         assertThat(builder.getNextLink(), is(nullValue()));
     }
-}
\ No newline at end of file
+    
+    @Test
+    public void shouldGenerateSelfLinkWithOnlyPrescribedSpecialCharactersURLEncoded(){
+        Map<String, Object> valuesForReferenceParameter = Map.of(
+                "{ref", "reference=%7Bref",
+                "[ref", "reference=%5Bref",
+                "r{e{f}{", "reference=r%7Be%7Bf%7D%7B",
+                "ref=", "reference=ref=",
+                "ref&", "reference=ref&",
+                "ref@", "reference=ref%40"
```

Comment:
A test with a space would be good, given they can become `%20` or `+` depending on criteria I cannot quite remember

Reply:
Cool, I'll add that in too 👍